### ISSUE TITLE
Add tidyr verbs (spread, gather, etc.)

### ISSUE THREAD

**Issue by alexkyllo** [cloudyr/AzureKusto#8]
We should be able to enhance this to add support for the verbs from [tidyr](https://tidyr.tidyverse.org/reference/index.html). 

I think that at least `extract`,  `spread`, `gather`, `separate`, `separate_rows` and `unite` should be doable based on my understanding of Kusto functionality. 

`nest` and `unnest` might not be possible because they involve data frames of data frames, and I don't think Kusto can return a table of tables.

**Comment by Hong-Revo** [comment]
Sure, if you want to have a crack at it. I haven't used the tidyr verbs much myself, so I can't really say much about them.

**Comment by alexkyllo** [comment]
Yeah, it'd be a cool "bonus" feature. I use gather and spread pretty frequently in my own data analysis, as well as the equivalent PIVOT and UNPIVOT in T-SQL. These aren't even supported in dbplyr, though, because the different SQL vendors implement them so inconsistently.

I'm planning to also open enhancement issues for the remaining portions of the dplyr API, to use as my own to-do list.
